src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -90,6 +91,20 @@ class KeyResults:
     def __init__(self) -> None:
         self.results: MutableMapping[int, MutableMapping[str, int]] = defaultdict(dict)
 
+    @classmethod
+    def merge_results(
```

Comment:
This method is not merging `KeyResults` instances nor it uses any of the `KeyResult` functionalities. So it does not need to be in this class. Its responsibility is separate. I would suggest you just make it a stand alone function.